http-servers/src/main/java/com/palantir/remoting/http/server/DropwizardTracingFilters.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting.http.server;
+
+import com.github.kristofa.brave.LoggingSpanCollector;
+import com.github.kristofa.brave.Sampler;
+import com.github.kristofa.brave.ServerRequestInterceptor;
+import com.github.kristofa.brave.ServerResponseInterceptor;
+import com.github.kristofa.brave.ServerTracer;
+import com.github.kristofa.brave.ThreadLocalServerClientAndLocalSpanState;
+import com.github.kristofa.brave.http.DefaultSpanNameProvider;
+import com.github.kristofa.brave.jaxrs2.BraveContainerRequestFilter;
+import com.github.kristofa.brave.jaxrs2.BraveContainerResponseFilter;
+import com.google.common.base.Function;
+import com.google.common.base.Optional;
+import com.google.common.base.Strings;
+import com.google.common.net.InetAddresses;
+import io.dropwizard.Configuration;
+import io.dropwizard.jersey.setup.JerseyEnvironment;
+import io.dropwizard.jetty.ConnectorFactory;
+import io.dropwizard.jetty.HttpConnectorFactory;
+import io.dropwizard.jetty.HttpsConnectorFactory;
+import io.dropwizard.server.DefaultServerFactory;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Random;
+
+/** Static utilities for registering Brave/Zipkin filters with Dropwizard applications. */
+public final class DropwizardTracingFilters {
+
+    private DropwizardTracingFilters() {}
+
+    /**
+     * Registers Brave request&response filters for logging Zipkin-style tracing information. The Zipkin logger carries
+     * the given tracerName and logs server IP/Port information extracted from the given Dropwizard configuration with
+     * "best effort".
+     * <p>
+     * TODO(rfink) Is there a more stable way to retrieve IP/Port information?
+     */
+    public static void registerBraveTracers(JerseyEnvironment environment, Configuration config, String tracerName) {
```

Comment:
done

On Wed, Jun 22, 2016 at 7:57 PM Mark Elliot <notifications@github.com>
wrote:

> In
> http-servers/src/main/java/com/palantir/remoting/http/server/DropwizardTracingFilters.java
> <https://github.com/palantir/http-remoting/pull/142#discussion_r68169166>:
>
> > +import java.net.UnknownHostException;
> > +import java.util.Random;
> > +
> > +/** Static utilities for registering Brave/Zipkin filters with Dropwizard applications. */
> > +public final class DropwizardTracingFilters {
> > +
> > +    private DropwizardTracingFilters() {}
> > +
> > +    /**
> > +     * Registers Brave request&response filters for logging Zipkin-style tracing information. The Zipkin logger carries
> > +     * the given tracerName and logs server IP/Port information extracted from the given Dropwizard configuration with
> > +     * "best effort".
> > +     * <p>
> > +     * TODO(rfink) Is there a more stable way to retrieve IP/Port information?
> > +     */
> > +    public static void registerBraveTracers(JerseyEnvironment environment, Configuration config, String tracerName) {
>
> registerTracers
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/palantir/http-remoting/pull/142/files/7dda2eb2f1c7d692e4d6aea670af8f34ac4042c4#r68169166>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AGOdwXaU1IMzVLTbQYW8XAJcv9c_DNkyks5qOfYBgaJpZM4I8W6U>
> .
>